packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/PropertySerializationProvider.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Generator.CSharp.Input;
+
+namespace Microsoft.Generator.CSharp.Providers
+{
+    public class PropertySerializationProvider
```

Comment:
Lets maybe call this `WireInfo` (not to fond of the name so open to suggestoins).

Reason being is this really represents the service contract of how this this property looks on the wire and whether it must be there or can be nullable.

It happens to be used in the serialization logic, but its not directly serialization information.

I also think this should just be an output type minus the provider namespace / suffix as its not a csharp concept like a Type, Method, Parameter, Property, etc.